app/src/main/java/io/apicurio/registry/content/canon/XmlContentCanonicalizer.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2020 JBoss Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.apicurio.registry.content.canon;
+
+import java.io.IOException;
+import javax.xml.parsers.ParserConfigurationException;
+import org.apache.xml.security.Init;
+import io.apicurio.registry.utils.IoUtil;
+import org.apache.xml.security.c14n.CanonicalizationException;
+import org.apache.xml.security.c14n.Canonicalizer;
+import org.apache.xml.security.c14n.InvalidCanonicalizerException;
+import org.xml.sax.SAXException;
+import io.apicurio.registry.content.ContentCanonicalizer;
+import io.apicurio.registry.content.ContentHandle;
+
+/**
+ * @author cfoskin@redhat.com
+ */
+
+public class XmlContentCanonicalizer implements ContentCanonicalizer {
+
+    /**
+     * @see io.apicurio.registry.content.ContentCanonicalizer#canonicalize(io.apicurio.registry.content.ContentHandle)
+     */
+    @Override
+    public ContentHandle canonicalize(ContentHandle content) {
+        Init.init(); // this library needs to be initialized using the static init()
+        try {
+            Canonicalizer canon = Canonicalizer.getInstance(Canonicalizer.ALGO_ID_C14N_OMIT_COMMENTS);
```

Comment:
Check the javadoc for `Canonicalizer` and see if it is threadsafe.  If it is, then we can create just one of these and re-use it each time this method is called.